python_modules/dagster/dagster/_core/definitions/asset_graph_computation.py

```diff
@@ -34,7 +38,6 @@ class AssetGraphComputation(IHaveNew):
     is_subset: bool
     selected_asset_keys: AbstractSet[AssetKey]
     selected_asset_check_keys: AbstractSet[AssetCheckKey]
-    output_names_by_key: Mapping[AssetKey, str]
```

Comment:
Previously, this was a property that was excluded from the constructor so that it could be derived from `keys_by_output_name`. But having a property that's not an argument to the constructor makes it impossible to use `copy`, which one of the transferred methods in this PR depends on.

So, in this PR, I instead made it a `cached_property` and forced it to be computed in the constructor.